src/engine_constants/mod.rs

```diff
@@ -1832,7 +1921,25 @@ impl EngineConstants {
         Ok(())
     }
 
-    pub fn apply_constant_json_files(&mut self) {}
+    pub fn apply_constant_json_files(&mut self, ctx: &mut Context) -> GameResult {
+        self.load_music_table(ctx)?;
+        Ok(())
+    }
+
+    pub fn load_music_table(&mut self, ctx: &mut Context) -> GameResult {
+        if let Ok(file) = filesystem::open_find(ctx, &self.base_paths, "music_table.json") {
```

Comment:
if loading fails, we should use the built-in file and show a warning to the user

we should add a `&self.builtin_paths` that only contains the special built-in base directory which can be used in fallback scenarios like this (it stops open_find from reading from all data directories but built-in)